quesma/quesma/functionality/bulk/bulk.go

```diff
@@ -104,16 +104,23 @@ func Write(ctx context.Context, defaultIndex *string, bulk types.NDJSON, ip *ing
 	}
 
 	if ip != nil {
-		sendToClickhouse(ctx, clickhouseDocumentsToInsert, phoneHomeClient, ingestStatsEnabled, ip)
+		sendToClickhouse(ctx, clickhouseBulkEntries, phoneHomeClient, ingestStatsEnabled, ip)
 	}
 
-	return results, nil
+	// Here we filter out empty results so that final response does not contain empty elements
+	nonEmptyResults := make([]BulkItem, 0, len(results))
+	for _, result := range results {
+		if result != (BulkItem{}) {
+			nonEmptyResults = append(nonEmptyResults, result)
+		}
+	}
+	return nonEmptyResults, nil
```

Comment:
alternatively `SplitBulk` could return properly-sized `results`

Reply:
Yea, as much as I don't like the current approach with shared array flying around between methods, I purposefully  avoided changing that in this PR because of the amount of changes I had to make so far.

I'll place a little TODO though.